### PR TITLE
NxAccordion in nx-tile - RSC-394

### PR DESCRIPTION
**Pull request by rpokorny**
https://issues.sonatype.org/browse/RSC-394

image

Note that there were a number of old, defunct nx-tile styles that I removed rather than having to check whether they still "work" with these changes.

### CODE REVIEW

File: lib/src/base-styles/_nx-tile.scss [hunk 1]
```diff
@@ -9,6 +9,10 @@ $nx-tile-header-height: 24px; // Minimum height
 $nx-tile-header-height-actions: 43px; // Height with sub-title and/or right actions
 $nx-tile-background-color: $nx-white;
 
+// the "padding" around the inside edge of the nx-tile, which must be implemented using padding/margin
+// on children since nx-tile-content--accordion-container is supposed to go all the way to the edges
+$nx-tile-padding: $nx-spacing-l;
```
Comment (glenhunter) [inline lib/src/base-styles/_nx-tile.scss:14]: You've created this variable, but I can't see anywhere that you have used it.

**Comment by glenhunter** [comment]
I noticed that you've purged a bunch of unused tile styling, I'm not opposed to that (despite being outside the remit of this story), but I think you should add an explanatory note to your PR description.

**Comment by rpokorny** [comment]
> I noticed that you've purged a bunch of unused tile styling, I'm not opposed to that (despite being outside the remit of this story), but I think you should add an explanatory note to your PR description.

Note added, it explains how I think those changes are relevant to this PR.

File: lib/src/base-styles/_nx-tile.scss (hunk 1 above)
Comment (rpokorny) [reply lib/src/base-styles/_nx-tile.scss:14]: > You've created this variable, but I can't see anywhere that you have used it.

Whoops. e6733595

File: gallery/src/styles/NxTile/NxTileAccordionExample.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      </section>
+      <section className="nx-tile">
+        <header className="nx-tile-header">
+          <div className="nx-tile-header__title">
+            <h2 className="nx-h2">Tile with normal content and an accordion</h2>
+          </div>
+        </header>
+        <div className="nx-tile-content">
+          <p className="nx-p">
+            Other content may optionally be present outside of the accordion container
```
Comment (glenhunter) [inline gallery/src/styles/NxTile/NxTileAccordionExample.tsx:90]: To me the key take-away here isn't that there can be content outside the accordion, but pointing out that an `nx-tile` can have multiple `nx-tile-content`'s, and that one could be "normal" and one could be for an accordion. Using a normal `nx-tile-content` and another with the accordion modifier is just a great way to demonstrate that capability.

Comment (rpokorny) [reply gallery/src/styles/NxTile/NxTileAccordionExample.tsx:90]: > To me the key take-away here isn't that there can be content outside the accordion, but pointing out that an nx-tile can have multiple nx-tile-content's, and that one could be "normal" and one could be for an accordion. Using a normal nx-tile-content and another with the accordion modifier is just a great way to demonstrate that capability.

Is there a particular change you are requesting? I feel like between this paragraph and the changes to the table in `NxTilePage` I've captured what you're saying.

Comment (glenhunter) [inline gallery/src/styles/NxTile/NxTileAccordionExample.tsx:90]:
```suggestion
            An <code className="nx-code">nx-tile</code> may have multiple <code className="nx-code">nx-tile-content</code> containers so an accordion can appear within a tile with "normal" content.
```

Comment (rpokorny) [reply gallery/src/styles/NxTile/NxTileAccordionExample.tsx]: > ```
> -             Other content may optionally be present outside of the accordion container
> +             An <code className="nx-code">nx-tile</code> may have multiple <code className="nx-code">nx-tile-content</code> containers so an accordion can appear within a tile with "normal" content.
> ```

Accepted with minor adjustments in f62a04a6

**Review by glenhunter — APPROVED** [review]

**Pull request merged**